ldms/src/ldmsd/ldmsd_stream.c

```diff
@@ -427,23 +427,60 @@ int ldmsd_stream_publish_file(const char *stream, const char *type,
 		goto err;
 	}
 
-	fseek(file, 0L, SEEK_END);
-	data_len = ftell(file);
-	rewind(file);
 	msg_no = ldmsd_msg_no_get();
+	rc = fseek(file, 0L, SEEK_END);
+	if (!rc) {
+		data_len = ftell(file);
+		rewind(file);
 
-	rc = stream_hdr_send(x, msg_no, stream, stream_type,
-			     buf, data_len + 1 /* terminating '\0' */);
-	if (rc)
-		goto close_xprt;
+		rc = stream_hdr_send(x, msg_no, stream, stream_type,
+				     buf, data_len + 1 /* terminating '\0' */);
+		if (rc)
+			goto close_xprt;
 
-	while ((cnt = fread(buffer, 1, max_msg_len-1, file)) > 0) {
-		if (cnt < max_msg_len-1) {
-			/* Ensure last buffer is '\0' terminated */
-			buffer[cnt] = '\0';
-			cnt += 1;
+		while ((cnt = fread(buffer, 1, max_msg_len-1, file)) > 0) {
+			if (cnt < max_msg_len-1) {
+				/* Ensure last buffer is '\0' terminated */
+				buffer[cnt] = '\0';
+				cnt += 1;
+			}
+			rc = stream_send(x, buf, msg_no, 0, buffer, cnt);
+			if (rc)
+				goto close_xprt;
 		}
-		rc = stream_send(x, buf, msg_no, 0, buffer, cnt);
+
+	} else {
+		if (ESPIPE != errno) {
+			msglog("The given file is invalid.\n");
+			goto close_xprt;
+		}
+
+		/* a non-seekable FILE */
+		char *s;
+		char b[max_msg_len];
```

Comment:
This buffer needs to be malloc'd to avoid blowing out the stack.

Reply:
Got it!